drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -269,6 +277,8 @@ static int stm32_clock_control_init(struct device *dev)
 	#error "CONFIG_CLOCK_STM32_SYSCLK_SRC_PLL not selected"
 #endif /* CLOCK_STM32_SYSCLK_SRC_PLL */
 
+	z_stm32_hsem_release(CFG_HW_RCC_SEMID);
```

Comment:
In line 269 `config_bus_prescalers()`, which takes the lock, is being invoked while the lock is held.  Is that correct (i.e. does this lock count)?

Reply:
This is not a problem because we use 1step lock mechanism which lock again CoreID only (compare current coreID with the one which may have lock the semaphore).

Reply:
So the `z_stm32_hsem_release()` invoked at the end of `config_bus_prescalers()` will also be ignored?  Or will the lock have been released, so the `z_stm32_hsem_release()` called here will have no effect?

Reply:
Humm ... right, semaphore will be release at first occurence of `z_stm32_hsem_release(`), which is not what is expected.
I get ride of `config_bus_prescalers()`  which was excusively used here, and thus only 1 Lock/release for the whole init processing.
Thanks for your accurate review.